packages/vite/src/node/plugins/css.ts

```diff
@@ -2108,7 +2128,58 @@ const makeScssWorker = (
         // eslint-disable-next-line no-restricted-globals -- this function runs inside a cjs worker
         const sass: typeof Sass = require(sassPath)
         // eslint-disable-next-line no-restricted-globals
-        const path = require('node:path')
+        const path: typeof import('node:path') = require('node:path')
+
+        if (options.api === 'modern') {
```

Comment:
Is it possible to move this new modern code out as a separate `makeModernScssWorker` function? This feels quite packed. This can also help make way when we support `modern-compiler`.

Reply:
Good point :+1:  I thought I can reuse current worker but it turned out it needs a different `parentFunctions`, so it's probably cleaner to split it out. The new one looks like this https://github.com/vitejs/vite/pull/17728/commits/7338db42add4052863a598029a79c58f236c1f7f